### PR TITLE
Add test update interval for vuln detector providers

### LINKED ISSUE #683: Test update_interval parameter from vuln detector providers
Hello team,

We want to test the attribute `update_interval` of Vulnerability detector feeds.

This test is related to #661 

Regards.

### PR DESCRIPTION
**Pull request by jmv74211**
Hello team,

This PR adds what is necessary to test the `update_interval` provider parameter for vulnerability detector.

The testing is done by checking the debug events that are stored in the `ossec.log`.

Closes #683

### Tests logic

After passing the time set in the `update_interval` parameter of the providers of vuln detector, it has to start downloading the provider's feeds and updating the database. For instance:

```
Starting National Vulnerability Database database update.
The update of the National Vulnerability Database feed finished successfully.
```

The test checks that the database download and update does not take place before the established time, and then the time advances and checks if the event occurs after passing the established interval.

This check is done for each provider `['redhat', 'canonical', 'debian', 'nvd']` and the following time units `['60s', '60m', '1h', '1d']`.

The number of total tests is **16** *(nProviders * nTimeUnits)*

> *Important*: The interval check is performed every x time set in the general interval parameter

### Tests checks

- [x] Proven that tests **pass** when they have to pass
- [x] Proven that tests **fail** when they have to fail
- [x] Proven that tests have the expected behavior in **RPM and DEB**
- [x] Checked that **all vulnerability detector tests work correctly**(my changes don't break anything)

#### RPM manager

- When the expected behavior occurs:

```
========================================= test session starts ==========================================
platform linux -- Python 3.6.8, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /root/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: metadata-1.8.0, html-2.0.1, testinfra-5.0.0
collected 16 items

test_vulnerability_detector/test_providers/test_update_interval.py ................              [100%]

==================================== 16 passed in 189.21s (0:03:09) ====================================
```

- When the module fails and does not start the update after the interval set:

```
========================================= test session starts ==========================================
platform linux -- Python 3.6.8, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /root/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: metadata-1.8.0, html-2.0.1, testinfra-5.0.0
collected 16 items

test_vulnerability_detector/test_providers/test_update_interval.py FFFFFFFFFFFFFFFF              [100%]

==================================== 16 failed in 363.73s (0:06:03) ====================================
```

#### DEB manager

- When the expected behavior occurs:

```
========================================= test session starts ==========================================
platform linux -- Python 3.8.2, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /root/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: html-2.0.1, testinfra-5.0.0, metadata-1.8.0
collected 16 items

test_vulnerability_detector/test_providers/test_update_interval.py ................              [100%]

==================================== 16 passed in 184.47s (0:03:04) ====================================
```

- When the module fails and does not start the update after the interval set:

```
========================================= test session starts ==========================================
platform linux -- Python 3.8.2, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /root/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: html-2.0.1, testinfra-5.0.0, metadata-1.8.0
collected 16 items

test_vulnerability_detector/test_providers/test_update_interval.py FFFFFFFFFFFFFFFF              [100%]
==================================== 16 failed in 354.15s (0:05:54) ====================================
```
#### Integrity check with other tests

```
========================================= test session starts ==========================================
platform linux -- Python 3.8.2, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /root/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: html-2.0.1, testinfra-5.0.0, metadata-1.8.0
collected 51 items

test_vulnerability_detector/test_general_settings/test_general_settings_enabled.py .ss.          [  7%]
test_vulnerability_detector/test_general_settings/test_general_settings_interval.py ............ [ 31%]
                                                                                                 [ 31%]
test_vulnerability_detector/test_general_settings/test_general_settings_run_on_start.py ..       [ 35%]
test_vulnerability_detector/test_providers/test_providers_enabled.py s                           [ 37%]
test_vulnerability_detector/test_providers/test_providers_update_from_year.py .........FF.....   [ 68%]
test_vulnerability_detector/test_providers/test_update_interval.py ................              [100%]

========================= 2 failed, 46 passed, 3 skipped in 249.98s (0:04:09) ==========================
```
> Note: The two failures obtained are expected errors.

Best regards

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/tools/monitoring.py [hunk 1]
```diff
@@ -373,7 +373,7 @@ def get_results(self, callback=_callback_default, accum_results=1, timeout=-1, u
                 else:
                     item = callback(self._queue.peek(position=position, block=True, timeout=self._time_step))
                     position += 1
-                if item is not None:
+                if item is not None and item is not False:
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/tools/monitoring.py:376]:
```suggestion
                if item is not None and item:
```

File: deps/wazuh_testing/wazuh_testing/tools/time.py [hunk 2]
```diff
@@ -262,3 +262,33 @@ def unit_to_seconds(time_):
...
+    if time_unit == "s":
+        return time_value
+    elif time_unit == "m":
+        return time_value * 60
+    elif time_unit == "h":
+        return time_value * 3600
+    elif time_unit == "d":
+        return time_value * 86400
+    elif time_unit == "w":
+        return time_value * 604800
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/tools/time.py:294]: Use a dictionary containing the values for each unit, and you can simplify this block to:

```suggestion
units = { 's': 1, 'm': 60, ... }
return time_value * units[time_unit]
```

File: deps/wazuh_testing/wazuh_testing/tools/monitoring.py (hunk 1 above)
Comment (jmv74211) [reply deps/wazuh_testing/wazuh_testing/tools/monitoring.py]: Done in b3d023b38d8c6abac7fb3757a859b4b3ba4574d9

File: deps/wazuh_testing/wazuh_testing/tools/time.py (hunk 2 above)
Comment (jmv74211) [reply deps/wazuh_testing/wazuh_testing/tools/time.py]: Done in b3d023b38d8c6abac7fb3757a859b4b3ba4574d9

**Pull request merged**